### PR TITLE
Cli version header

### PR DESCRIPTION
**Pull request by ryan-cahill**

### CODE REVIEW

File: src/app-config/service.ts [hunk 1]
```diff
@@ -91,6 +93,7 @@ export default class AppService {
       const { token_type, access_token } = this.auth.auth_results;
       this._api.defaults.headers = {
         Authorization: `${token_type} ${access_token}`,
+        'Cli-Version': this.version,
```
Comment (dan-barrett) [inline src/app-config/service.ts:96]: Should this be `X-Cli-Version` since this is a custom header?

File: test/commands/build.test.ts [hunk 2]
```diff
@@ -64,7 +64,7 @@ describe('build', function () {
     });
     const tmp_config_file = path.join(tmp_dir, ARCHITECTPATHS.CLI_CONFIG_FILENAME);
     fs.writeJSONSync(tmp_config_file, config);
-    const app_config_stub = sinon.stub().resolves(new AppService(tmp_dir));
+    const app_config_stub = sinon.stub().resolves(new AppService(tmp_dir, '0.0.1'));
```
Comment (dan-barrett) [inline test/commands/build.test.ts:67]: Will these need to be changed as the CLI gets updated or is `0.0.1` more of a "mock version" for testing purposes only?

My only comment would be that we consolidate these references if we're having to update them frequently, but I think this pattern is fine if they stay `0.0.1` forever.

File: src/base-command.ts [hunk 3]
```diff
@@ -15,13 +15,9 @@ export default abstract class extends Command {
...
   };
 
-  private getFormattedJSON(obj: object) {
-    return JSON.stringify(obj, null, 2);
-  }
-
   async init() {
     if (!this.app) {
-      this.app = await AppService.create(this.config.configDir);
+      this.app = await AppService.create(this.config.configDir, this.config.userAgent.split(/\/|\s/g)[2]);
```
Comment (dan-barrett) [inline src/base-command.ts:20]: Nice 👍 , presumably this will grab this: `0.4.10-rc.1`?

Worth sending the entire userAgent in another header (eg. `@architect-io/cli/0.4.10-rc.1 darwin-x64 node-v13.3.0`) for analytics purposes? scope creep...

File: test/commands/build.test.ts (hunk 2 above)
Comment (ryan-cahill) [reply test/commands/build.test.ts:67]: I don't really see why we would need to change them since these are just tests, and this isn't related to any of the tests

File: src/base-command.ts (hunk 3 above)
Comment (ryan-cahill) [reply src/base-command.ts:20]: I had done that originally, but didn't really think the extra information was that useful. When we do, maybe we can send it to Segment or something, but that data currently has no place to go anyway

File: src/app-config/service.ts (hunk 1 above)
Comment (ryan-cahill) [reply src/app-config/service.ts:96]: I had been unsure about that at first, but sources say the `X-` prefix is basically deprecated

https://specs.openstack.org/openstack/api-wg/guidelines/headers.html

Comment (dan-barrett) [reply src/app-config/service.ts:96]: Wow... would you look at that! Good call, and good to know.

File: src/base-command.ts (hunk 3 above)
Comment (dan-barrett) [reply src/base-command.ts:20]: Fair enough... this is me just being greedy. Agreed, probably not worth it for now.

Will be useful someday when we encounter issues that we can't reproduce and want to know environment details, but we're probably a long ways away from that right now since we can just ask the friendly what their userAgent is.

**Comment by dan-barrett** [comment]
@tjhiggins @ryan-cahill approved from me
